docs/quickstarts/fastify.mdx

```diff
@@ -27,171 +65,162 @@ After following this guide, you should have a working Fastify app with public an
 
-  ### Configure `clerkPlugin` in your Fastify application
-
-  The Clerk plugin can be registered globally or for specific routes. This examples registers the plugin globally.
+  > [!IMPORTANT]
+  > The `dotenv/config` module must be imported before any Clerk modules. This order is important because Clerk instances are created during the import process and rely on environment variables, such as API keys, to be initialized correctly. For more details, refer to the [Fastify docs](https://fastify.dev/docs/latest/Guides/Getting-Started/#loading-order-of-your-plugins).
 
   ```ts {{ filename: 'index.ts' }}
-  import * as dotenv from 'dotenv'
-
-  dotenv.config()
-
+  import 'dotenv/config'
   import Fastify from 'fastify'
-  import { clerkClient, clerkPlugin, getAuth } from '@clerk/fastify'
+  import { clerkPlugin } from '@clerk/fastify'
 
   const fastify = Fastify({ logger: true })
 
   fastify.register(clerkPlugin)
 
   const start = async () => {
     try {
-      await fastify.listen({ port: 3000 })
-    } catch (err) {
-      fastify.log.error(err)
+      await fastify.listen({ port: 8080 })
+    } catch (error) {
+      fastify.log.error(error)
       process.exit(1)
     }
   }
 
   start()
   ```
 
-  ### Accessing auth state using `getAuth`
+  ### Use `getAuth()` to access the current auth state
 
-  The `getAuth` helper can be used to access the auth state of the current request.
-
-  ```ts {{ filename: 'index.ts' }}
-  import * as dotenv from 'dotenv'
-
-  dotenv.config()
+  The `getAuth()` method can access the authentation state of the current request. In the following example, the `userId` loads the entire [`User`](/docs/references/javascript/user/user) object from Clerk.
 
+  ```ts {{ filename: 'index.ts', mark: [3, [9, 17]] }}
+  import 'dotenv/config'
   import Fastify from 'fastify'
   import { clerkClient, clerkPlugin, getAuth } from '@clerk/fastify'
 
   const fastify = Fastify({ logger: true })
 
   fastify.register(clerkPlugin)
 
-  fastify.get('/', async (req, reply) => {
-    /**
-     * Access the auth state for this request.
-     * In this example, the userId loads the whole User object
-     * from the Clerk servers
-     */
-    const { userId } = getAuth(req)
+  fastify.get('/', async (request, reply) => {
+    const { userId } = getAuth(request)
     const user = userId ? await clerkClient.users.getUser(userId) : null
-    return { user }
+
+    return reply.send({
+      message: 'Authentication state retrieved successfully.',
+      user,
+    })
   })
 
   const start = async () => {
     try {
-      await fastify.listen({ port: 3000 })
-    } catch (err) {
-      fastify.log.error(err)
+      await fastify.listen({ port: 8080 })
+    } catch (error) {
+      fastify.log.error(error)
       process.exit(1)
     }
   }
 
   start()
   ```
 
-  ### Require authentication for a route
+  ### Use `getAuth()` to protect specific routes
```

Comment:
This section can be combined with the one above. See how its done [here](https://github.com/clerk/clerk-docs/pull/1201/files#diff-b704145027aa89795fd84c206c02c3f07082daaa6be1549c2f531aede6330774R112)